idea/src/org/jetbrains/kotlin/idea/quickfix/ChangeVariableMutabilityFix.kt

```diff
@@ -86,15 +88,13 @@ class ChangeVariableMutabilityFix(
 
         val CAPTURED_MEMBER_VAL_INITIALIZATION_FACTORY = ReassignmentActionFactory(Errors.CAPTURED_MEMBER_VAL_INITIALIZATION)
 
-        val VAR_OVERRIDDEN_BY_VAL_FACTORY: KotlinSingleIntentionActionFactory = object : KotlinSingleIntentionActionFactory() {
-            override fun createAction(diagnostic: Diagnostic): IntentionAction? {
-                val element = diagnostic.psiElement
-                return when (element) {
-                    is KtProperty, is KtParameter -> ChangeVariableMutabilityFix(element as KtValVarKeywordOwner, true)
-                    else -> null
+        val VAR_OVERRIDDEN_BY_VAL_FACTORY: QuickFixesPsiBasedFactory<PsiElement> =
```

Comment:
Please move the whole file into `idea-front-end-independent` module as it is used now by `idea-fir`. We a trying to keep dependencies to FE1.0 plugin classes to the minimum. Unfortunately, we can not remove the dependencies to the  FE1.0 plugin modules right now, but I hope we can do it soon

Reply:
`ChangeVariableMutabilityFix` inherits from `KotlinQuickFixAction` which has a dependency on `org.jetbrains.kotlin.idea.caches.resolve.allowCachedResolveInDispatchThread` which is in the `idea` module. `KotlinQuickFixAction` can't simply be moved to `idea-frontend-independent` as it will be a circular dependency.
It seems fine to remove this call for `ChangeVariableMutabilityFix` (no resolve required) but not for some of the other actions that extend `KotlinQuickFixAction`. I created a different version `KotlinPsiOnlyQuickFixAction` that doesn’t have this call, for fixes that don't require resolve.